### DISCUSSION TITLE
Provide a pure web/JS wallet based on BIP 39

### DISCUSSION THREAD

**Discussion by tjanez** [oasisprotocol/community-forum#2] [Ideas]
##### SUMMARY

Consider providing a pure web/JS wallet that would support private key/account derivation from a [BIP 39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) mnemonic.

A good example would be how [Avalanche](https://avax.network) did it on the mnemonic and private key management side: https://wallet.avax.network/.

_Originally suggested by @fmanea in https://github.com/oasisprotocol/community/issues/1#issuecomment-723349908 and https://github.com/oasisprotocol/community/issues/1#issuecomment-723827378._

/cc @Bambarello

**Comment by expyri** [comment]
right !

**Comment by Esya** [comment]
I'm interested in building this.

Would the following general architecture be OK ? : 

- Deploy a non-validator node along with an oasis-rosetta-gateway somewhere
- Use the construction API from this rosetta gateway, but sign everything locally
- Submit the transactions to the rosetta gateway

First release would have : 

- Wallet creation / recovery from mnemonic phrase or private key
- Account balance
- Transaction history
- Sending/receiving tokens
- Delegation

I think I see no issue moving forward, it's just a matter of time and availability to build this. I'd also like to know if this would possibly be eligible for the grant program - I'm pretty sure this could help Oasis gain traction.

**Comment by mrtaspinar** [comment]
Hi ,
when it will be active , please inform.
I wish you a happy work day

**Reply by tjanez** [reply]
The code repo is here: https://github.com/Esya/oasis-wallet

The preliminary version for the Testnet is here: https://testnet.oasis-wallet.com/

**Comment by tjanez** [comment]
This has been done.

The official Oasis Wallet, the web variant, has been launched on Sep 22, 2021.

To use it, browse to: https://wallet.oasisprotocol.org/

More details:
- Documentation: https://docs.oasis.dev/general/manage-tokens/oasis-wallets
- Code: https://github.com/oasisprotocol/oasis-wallet-web
